source/Nevermore/Advanced/WriteTransaction.cs

```diff
@@ -23,20 +23,40 @@ public class WriteTransaction : ReadTransaction, IRelationalTransaction
         readonly IKeyAllocator keyAllocator;
         readonly DataModificationQueryBuilder builder;
 
+#nullable enable
         public WriteTransaction(
             IRelationalStore store,
-            RelationalTransactionRegistry registry,
+            IRelationalTransactionRegistry registry,
             RetriableOperation operationsToRetry,
             IRelationalStoreConfiguration configuration,
             IKeyAllocator keyAllocator,
-            string name = null
-        ) : base(store, registry, operationsToRetry, configuration, name)
+            Action<string>? customCommandTrace = null,
```

Comment:
is this unique to each ctor invocation, or could it be on the config object?

Reply:
In theory you _could_ put it on the config object and for some simple use cases that might be sufficient, but in practice it's so much more convenient if you can directly associate the tracing with the specific transaction that the trace is coming from by just binding them together like this.

Another nice property from doing it this way is it's optional per Nevermore transaction; for externally managed transactions this is really useful, but if you're not adopting an existing transaction then it's less useful, so in a codebase that does a mix of both you might only want custom tracing in some cases and can just pass a `null` in the others.